src/osd/scrubber/pg_scrubber.cc

```diff
@@ -216,9 +222,12 @@ void PgScrubber::initiate_regular_scrub(epoch_t epoch_queued)
   }
 }
 
-void PgScrubber::dec_scrubs_remote()
+void PgScrubber::clear_reservation_by_remote_primary()
 {
-  m_osds->get_scrub_services().dec_scrubs_remote();
+  m_osds->get_scrub_services().dec_scrubs_remote(m_pg_id.pgid);
```

Comment:
This is already handled by ReplicaActive::~ReplicaActive.  Interestingly, advance_token() below is not.
1. Add advance_token() to ReplicaActive::~ReplicaActive
2. Remove every other instance of calling dec_scrubs_remote or clear_reservation_by_remote_primary

Reply:
@athanatos
Handled the dec_scrubs...().
Regarding advance_token() - that one required a fix, as its meaning was slightly modified by the PR. I have made the required changes.
Also - added a related commit to handle the (unexpected) case of a new chunk request (with a newly assigned token) arriving while we are still handling the previous one. Handled here, as otherwise - the scrubber will be stuck.